lib/checkPullRequestReview.js

```diff
@@ -0,0 +1,292 @@
+  const reviewApprovedSearchString = 'review:approved';
+  const searchResult = await context.github.search.issuesAndPullRequests(
+    context.repo({
+      q:
+        repoSearchString +
+        ' ' +
+        reviewApprovedSearchString +
+        ' ' +
+        pullRequest.number,
+    })
+  );
+
+  const searchData = searchResult.data.items;
+  const assignees = pullRequest.assignees.map((assignee) => assignee.login);
+  if (searchData.length === 0) {
+    // Pull request has not been approved by all reviewers, hence we need to
+    // assign the remaining reviewers.
+    const reviewersYetToReview = pullRequest.requested_reviewers.map(
+      (reviewer) => reviewer.login
+    );
+
+    const unassignedReviewers = reviewersYetToReview.filter(
+      (reviewer) => !assignees.includes(reviewer)
+    );
+    if (unassignedReviewers.length > 0) {
+      // Ping and assign unassigned reviewers.
+      const commentBody =
+        'Assigning @' +
+        unassignedReviewers.join(', @') +
+        ' for code owner reviews, Thanks!';
+
+      await context.github.issues.createComment(
+        context.repo({
+          issue_number: pullRequest.number,
+          body: commentBody,
+        })
+      );
+
+      await context.github.issues.addAssignees(
+        context.repo({
+          issue_number: pullRequest.number,
+          assignees: unassignedReviewers,
+        })
+      );
+    }
+  } else {
+    // Since the pull request has been approved, we add the LGTM label.
+    const labels = pullRequest.labels.map((label) => label.name);
+    if (!labels.includes(LGTM_LABEL)) {
+      await context.github.issues.addLabels(
+        context.repo({
+          issue_number: pullRequest.number,
+          labels: [LGTM_LABEL],
+        })
+      );
+    }
+
+    // Check if author can merge PR. All members can merge the pull request.
+    const membershipCheckResponse = await context.github.orgs.checkMembership({
+      org: context.payload.organization.login,
+      username: pullRequest.user.login,
+    });
+    const SUCCESS_STATUS = 204;
+
+    if (membershipCheckResponse.status === SUCCESS_STATUS) {
+      // Assign author.
+      await context.github.issues.addAssignees(
+        context.repo({
+          issue_number: pullRequest.number,
+          assignees: [pullRequest.user.login],
+        })
+      );
+      await context.github.issues.createComment(
+        context.repo({
+          issue_number: pullRequest.number,
+          body:
+            'Hi @' +
+            pullRequest.user.login +
+            ', this PR is ready to be merged. Please make sure there are no ' +
+            'pending comments before merge. Thanks!',
+        })
+      );
+    } else {
+      // Assign project owner to merge the pull request.
+      // A pull request will always have a changelog label because we are
+      // assigning the PR author to the pull request on creation if created
```

Comment:
assigning the PR author to the pull request on creation if created without a changelog label  --> assigning the PR author to the pull request if it is created without changelog label

Reply:
Done.